fees/solv-finance/index.ts

```diff
@@ -50,6 +56,74 @@ const fetch: FetchV2 = async (options) => {
 
 
   const dailyFees = options.createBalances();
+  const feeRevenue = await feeRevenues(options, contracts);
+  dailyFees.addBalances(feeRevenue);
+
+  const dailyRevenue = options.createBalances();
+  const receivedRevenues = await revenues(options, contracts);
+  dailyRevenue.addBalances(receivedRevenues);
+
+  dailyRevenue.addBalances(dailyFees.clone(yields));
+
+  return {
+    dailyFees,
+    dailyRevenue,
+  };
+};
+
+async function revenues(options: FetchOptions, contracts: any): Promise<Balances> {
+  const dailyRevenues = options.createBalances();
+
+  const receivedRevenue = await receivedRevenues(options, contracts);
+  dailyRevenues.addBalances(receivedRevenue);
+
+  const nativeTokenRevenue = await nativeTokenRevenues(options, contracts);
+  dailyRevenues.addBalances(nativeTokenRevenue);
+
+  const solanaRevenue = await solanaRevenues(options, contracts);
+  dailyRevenues.addBalances(solanaRevenue);
+
+  return dailyRevenues;
+}
+
+async function receivedRevenues(options: FetchOptions, contracts: any): Promise<Balances> {
+  const receivedRevenueConfig = contracts[options.chain]?.receivedRevenue;
+  if (!receivedRevenueConfig) {
+    return options.createBalances();
+  }
+
+  return addTokensReceived({
+    options,
+    targets: receivedRevenueConfig.address,
+    tokens: receivedRevenueConfig.token,
+  });
+}
+
+async function nativeTokenRevenues(options: FetchOptions, contracts: any): Promise<Balances> {
+  const nativeTokenConfig = contracts[options.chain]?.nativeTokenRevenue;
+  if (!nativeTokenConfig) {
+    return options.createBalances();
+  }
+
+  return addGasTokensReceived({
+    multisig: nativeTokenConfig.address,
+    options,
+  });
+}
+
+async function solanaRevenues(options: FetchOptions, contracts: any): Promise<Balances> {
+  const solanaRevenueConfig = contracts[options.chain]?.solanaRevenue;
+  if (!solanaRevenueConfig) {
+    return options.createBalances();
+  }
+
+  return await getSolanaReceived({ options, targets: solanaRevenueConfig });
```

Comment:
will the tokens received as fees be sent to these wallets? wouldnt that lead to doublecount?